pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/readers/LuceneFSTIndexReader.java

```diff
@@ -52,8 +51,7 @@ public LuceneFSTIndexReader(PinotDataBuffer pinotDataBuffer)
     _dataBufferIndexInput = new PinotBufferIndexInput(_dataBuffer, 0L, _dataBuffer.size());
 
     _readFST =
-        new FST<>(FST.readMetadata(_dataBufferIndexInput, PositiveIntOutputs.getSingleton()),
-            _dataBufferIndexInput, new OffHeapFSTStore());
+        new FST<>(FST.readMetadata(_dataBufferIndexInput, PositiveIntOutputs.getSingleton()), _dataBufferIndexInput);
```

Comment:
In recent tests, I found OnHeapFSTStore got used by default
```
public FST(FSTMetadata<T> metadata, DataInput in) throws IOException {
    this(metadata, in, new OnHeapFSTStore(DEFAULT_MAX_BLOCK_BITS));
  }
```

I'm looking at how to create the reader with OffheapFSTStore? Feel free to chime in.

Reply:
https://github.com/apache/pinot/pull/14457